pkg/operation/care/constraints.go

```diff
@@ -152,6 +189,59 @@ func getMutatingWebhookConfigurations(ctx context.Context, c client.Client) ([]a
 	return mutatingWebhookConfigs.Items, nil
 }
 
+// CheckIfCACertificateValiditiesAcceptable checks whether there are CA certificates which are expiring in less than a
+// year.
+func (c *Constraint) CheckIfCACertificateValiditiesAcceptable(ctx context.Context) (gardencorev1beta1.ConditionStatus, string, string, []gardencorev1beta1.ErrorCode, error) {
+	// CA certificates are valid for 10y, so let's only consider those certificates problematic which are valid for less
+	// than 1y.
+	const minimumValidity = 24 * time.Hour * 365
+
+	secretList := &corev1.SecretList{}
+	if err := c.seedClient.List(ctx, secretList, client.InNamespace(c.shoot.SeedNamespace), client.MatchingLabels{
+		secretsmanager.LabelKeyManagedBy:       secretsmanager.LabelValueSecretsManager,
+		secretsmanager.LabelKeyManagerIdentity: v1beta1constants.SecretManagerIdentityGardenlet,
```

Comment:
I assume you have considered checking other CAs (e.g. managed by extensions) as well. So just for confirmation:
Probably it's not a good idea, because other managers might choose to use shorter validities and we would need to detect that as well (or use some kind of percentage threshold). In the end, this is anyways just a best-effort warning.
Is that your reasoning?

Reply:
I did not really think about it, but what you say sounds reasonable. In the end, I think it's rather a theoretical problem that will likely not occur in reality.

Reply:
Ok, then we are on the same page 👍